de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/dialog/UiSnippetRoleManagementHandleUsersListDialog.java

```diff
@@ -0,0 +1,384 @@
+	    GridData valueLabelGridData = new GridData(SWT.FILL, SWT.CENTER, false, false);
+	    valueLabelGridData.horizontalIndent = HORIZONTAL_INDENT_FOR_FEATURE_LABEL;  // Adjust the indentation
+	    valueLabel.setLayoutData(valueLabelGridData);
+	    Text valueText = new Text(container, SWT.BORDER);
+	    GridData valueTextGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    valueText.setLayoutData(valueTextGridData);
+	    // Features Label and List
+	    Composite featuresComposite = new Composite(container, SWT.NONE);
+	    featuresComposite.setLayout(new GridLayout());	    
+	    GridData featuresCompositeGridData = new GridData(SWT.FILL, SWT.FILL, true, true, 1, VERTICAL_SPAN);
+	    featuresComposite.setLayoutData(featuresCompositeGridData);
+	    Label featuresLabel = new Label(featuresComposite, SWT.NONE);
+	    featuresLabel.setText("List of users :");
+	    featuresLabel.setForeground(parent.getDisplay().getSystemColor(SWT.COLOR_BLUE));  // Set the foreground color
+	    GridData featuresLabelGridData = new GridData(SWT.FILL, SWT.CENTER, false, false);
+	    featuresLabelGridData.horizontalIndent = HORIZONTAL_INDENT_FOR_FEATURE_LABEL;  // Adjust the indentation
+	    featuresLabel.setLayoutData(featuresLabelGridData);
+	    featuresList = new List(featuresComposite, SWT.BORDER | SWT.V_SCROLL);
+	    GridData listGridData = new GridData(SWT.FILL, SWT.FILL, true, true);
+	    featuresList.setLayoutData(listGridData);
+	    featuresList.setItems(features);
+	    // Buttons
+	    Composite buttonsComposite = new Composite(container, SWT.NONE);
+	    buttonsComposite.setLayout(new GridLayout());
+	    GridData buttonsCompositeGridData = new GridData(SWT.FILL, SWT.FILL, false, false);
+	    buttonsComposite.setLayoutData(buttonsCompositeGridData);
+	    Button addButton = new Button(buttonsComposite, SWT.PUSH);
+	    addButton.setText("Add");
+	    GridData addButtonGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    addButton.setLayoutData(addButtonGridData);
+	    
+	    updateButton(buttonsComposite, valueText);
+	    
+	    Button removeButton = new Button(buttonsComposite, SWT.PUSH);
+	    removeButton.setText("Remove");
+	    GridData removeButtonGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    removeButton.setLayoutData(removeButtonGridData);
+	    Button upButton = new Button(buttonsComposite, SWT.PUSH);
+	    upButton.setText("Up");
+	    GridData upButtonGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    upButton.setLayoutData(upButtonGridData);
+
+	    Button downButton = new Button(buttonsComposite, SWT.PUSH);
+	    downButton.setText("Down");
+	    GridData downButtonGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    downButton.setLayoutData(downButtonGridData);  
+        // Implementation of listeners
+        downButton.addSelectionListener(new SelectionAdapter() {
```

Comment:
Maybe move all selection events to a new mthod as you did it with the update button in "updateButton"?

Reply:
i did it and created the method called addListeners().